src/hooks/use-list-behavior.js

```diff
@@ -0,0 +1,50 @@
+import { useCallback } from 'react'
+import safeInvoke from '../lib/safe-invoke'
+import { useRovingTabindex } from '../roving-tabindex'
+import { useClickable } from './use-clickable'
+
+export function useListBehavior({
+  disabled,
+  isHighlighted,
+  isMultiSelect,
+  isSelectable,
+  isSelected,
+  onSelect,
+  ref
+}) {
+  const { 'aria-selected': ariaSelected, onClick: rovingTabIndexOnClick, onKeyDown: rovingTabIndexOnKeyDown, tabIndex: rovingTabIndex, ...rovingProps } = useRovingTabindex({
+    disabled,
+    isSelectable,
+    isSelected,
+    ref
+  })
+
+  const { onKeyDown, tabIndex: useClickableTabIndex } = useClickable({
+    disabled,
+    onKeyDown: rovingTabIndexOnKeyDown,
+    tabIndex: rovingTabIndex
+  })
+
+  const clickHandler = useCallback(
+    event => {
+      safeInvoke(rovingTabIndexOnClick, event)
+      safeInvoke(onSelect, event)
+    },
+    [ onSelect, rovingTabIndexOnClick ]
+  )
+
+  const checkboxProps = {
+    role: isMultiSelect ? 'checkbox' : 'radio',
```

Comment:
For my own edification - the aria spec suggests if a list or combobox implements multiselect, then each item should have a role of checkbox? Otherwise radio?